### PR TITLE
Binary MNIST/EMNIST Datasets and Datamodules

### PR DESCRIPTION
**Pull request by matsumotosan**
## What does this PR do?

Fixes part of #839 
- `pl_bolts.datamodules.binary_emnist_datamodule.BinaryEMNISTDataModule`
- `pl_bolts.datamodules.binary_mnist_datamodule.BinaryMNISTDataModule`
- `pl_bolts.datasets.emnist_dataset.BinaryEMNIST` (add unit test)
- `pl_bolts.datasets.mnist_dataset.BinaryMNIST` (add unit test)

## Summary

- Add tests for binarized MNIST and EMNIST datasets
- Add threshold attribute (set to 127) to Dataset classes to be used for binarizing MNIST and EMNIST images
- Add tests for val and test dataloaders

By default, the images are returned as PIL images (mode='L') in line with the torchvision [MNIST](https://pytorch.org/vision/stable/_modules/torchvision/datasets/mnist.html#MNIST.__getitem__) class. The threshold value is manually set as of right now.

## Before submitting

- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-bolts/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does **only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes?
- [x] Did you write any **new necessary tests**? \[not needed for typos/docs\]
- [x] Did you verify new and **existing tests** pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-bolts/blob/master/CHANGELOG.md)?

## PR review

- [x] Is this pull request **ready for review**? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: pl_bolts/datasets/emnist_dataset.py [hunk 1]
```diff
@@ -14,9 +15,16 @@
...
-@under_review()
 class BinaryEMNIST(EMNIST):
-    def __getitem__(self, idx):
+    """Binarized EMNIST Dataset.
+
+    EMNIST dataset binarized using a thresholding operation. Threshold is set to 127. Note that the images are binarized
+    prior to the application of any transforms.
+    """
+
+    threshold = 127.0
```
Comment (otaj) [inline pl_bolts/datasets/emnist_dataset.py:25]: Can this threshold be an argument to `__init__`?

File: pl_bolts/datasets/mnist_dataset.py [hunk 2]
```diff
@@ -13,36 +14,21 @@
...
-@under_review()
 class BinaryMNIST(MNIST):
-    def __getitem__(self, idx):
+    """Binarized MNIST Dataset.
+
+    MNIST dataset binarized using a thresholding operation. Threshold is set to 127. Note that the images are binarized
+    prior to the application of any transforms.
+    """
+
+    threshold = 127.0
```
Comment (otaj) [inline pl_bolts/datasets/mnist_dataset.py:25]: Can this threshold be a part of the `__init__` call already?

**Review by otaj — APPROVED** [review]
Over all, it looks very good, I have just two nits left in the comments

File: pl_bolts/datasets/emnist_dataset.py (hunk 1 above)
Comment (matsumotosan) [reply pl_bolts/datasets/emnist_dataset.py]: Added threshold as an optional argument with default value as 127.

File: pl_bolts/datasets/mnist_dataset.py (hunk 2 above)
Comment (matsumotosan) [reply pl_bolts/datasets/mnist_dataset.py]: Added threshold as an optional argument with default value as 127.

**Pull request merged**